# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerDebuggerParent.cpp
# Commit: 1e774bd7f365
# Full Hash: 1e774bd7f36525a03b6260b1cd7741bcf4e5f622
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-04-09 04:12:49
# Regressor Bug: 1899503
# File Overlap Count: 4
# Description:
#   Bug 1899503 - P3 PRemoteWorkerDebugger binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebugger binding.
#   
#   We are trying to keep the same logic debugger registration/unregistration to minimize the difference between the local debugger(the debugger on the content process main thread)
# ==============================================================================

diff -r b003e26209cf -r 1e774bd7f365 dom/workers/remoteworkers/RemoteWorkerDebuggerParent.cpp
--- a/dom/workers/remoteworkers/RemoteWorkerDebuggerParent.cpp	Tue Apr 08 09:43:33 2025 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerDebuggerParent.cpp	Tue Apr 08 09:43:33 2025 +0000
@@ -4,6 +4,7 @@
 
 #include "RemoteWorkerDebuggerParent.h"
 #include "mozilla/dom/WorkerDebuggerManager.h"
+#include "mozilla/dom/WorkerPrivate.h"
 
 namespace mozilla::dom {
 
@@ -35,6 +36,10 @@
     listener->OnClose();
   }
 
+  if (CanSend()) {
+    Unused << SendUnregisterDone();
+  }
+
   return IPC_OK();
 }
 
@@ -106,6 +111,15 @@
 }
 
 NS_IMETHODIMP
+RemoteWorkerDebuggerParent::GetIsRemote(bool* aResult) {
+  AssertIsOnMainThread();
+  MOZ_ASSERT_DEBUG_OR_FUZZING(aResult);
+
+  *aResult = true;
+  return NS_OK;
+}
+
+NS_IMETHODIMP
 RemoteWorkerDebuggerParent::GetIsInitialized(bool* aResult) {
   AssertIsOnMainThread();
   MOZ_ASSERT_DEBUG_OR_FUZZING(aResult);